src/HDF5.jl

```diff
@@ -443,11 +443,10 @@ Pass `swmr=true` to enable (Single Writer Multiple Reader) SWMR write access for
 "r+", or SWMR read access for "r".
 """
 function h5open(filename::AbstractString, mode::AbstractString = "r"; swmr::Bool = false, pv...)
-    apl = create_property(H5P_FILE_ACCESS; pv...) # file access property list
     # With garbage collection, the other modes don't make sense
     # (Set this first, so that the user-passed properties can overwrite this.)
-    apl[:fclose_degree] = H5F_CLOSE_STRONG
-    cpl = create_property(H5P_FILE_CREATE; pv...) # file create property list
+    apl = create_property(H5P_FILE_ACCESS; fclose_degree = H5F_CLOSE_STRONG, pv...) # file access property list
+    cpl = isempty(pv) ? DEFAULT_PROPERTIES : create_property(H5P_FILE_CREATE; pv...) # file create property list
```

Comment:
There's also the question of setting `apl` properties versus what the comment says. The previous code sets `:fclose_degree` after constructing the property, so it's _not_ doing what the comment says, while here I switched it to match the comment. For comparison, the `_h5open` function above will always overwrite the `:fclose_degree` property.

I assume I should actually remove the comment and change back to always overriding `:fclose_degree`?

Reply:
Assuming the comment on GC is true, it sounds like we should always overwrite the  user-passed property for `fclose_degree`.

Reply:
Looks like I forgot to delete `_h5open` at some point.

Reply:
On second thought looking at https://github.com/JuliaIO/HDF5.jl/pull/617/files it looks like for virtual datasets we need a way to override  fclose_degree